local-cli/link/pods/findPodTargetLine.js

```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+module.exports = function findPodTargetLine(podLines, projectName) {
```

Comment:
Shouldn't this return an array? What if user has few targets in a single `pod` file, one e.g. for iOS and second for e.g. tvOS? Same issue is happening right now in standard `link` - we just link to first target rather than to all that include React. I am currently rewriting this approach.

Here, I believe we should find all targets that have `pod React` inside (they consume React Native via CocoaPods) and assume that if they already use React Native inside that target, they want to use linked module as well.

Reply:
@grabbou I can't agree more with you regarding our `link` issue where we just link the first target. Besides being able to have a target for different platforms (tvOS..) I think there are more use cases that'd use more than one target - a white label product (app) is a good example. My team face this issue every time we add a new library with native deps.. Let me know if I can help you in any way.

Reply:
This might be tricky to do. Current implementation is kind of naive, and works only for simple cases (that was assumptions from the beginning). So, I'm just looking for target that has same name as project, and then look for good place to add new line.
But to make it work in most cases, I would have to write some simple Ruby parser to look for all non-nested targets (possible only ones, that already had React as dependency?), and then add new line to them. And I'm sure, that there always will be some project that somehow abuse Podfile syntax, and my approach would fail.

Anyway, if you could send me some real life Podfiles, I might try to check few ways to make it better. Also if you have some ideas how to improve it, let me know.

Reply:
hmm, that's a good point, I don't think we'll be able to make it work in all podfiles.. (my comments above were more related to not cocoa-pod links..)
I have an example where we customized the Podfile so it'd be possible to add new dep for all targets at once.

```
[
  "Target1",
  "Target2",
  "Target3",
  "Target4",
  "Target5",
  "Target6",
].each do |target_name|
  target target_name do
    pod "IQKeyboardManager"
    pod "Firebase/Messaging"
    pod "Firebase/Crash"
    pod "Firebase/Core"
    pod "FLAnimatedImage"
  end
end
```

Reply:
Ok, so my implementation would not work for such file - I assumed that target name would be same as project name. Which might be true in this case, but regexp's are not smart enough to find it out :(

Reply:
You can check what Mobile Azure team has done - when you attemp to link one of their native libraries, they ask you to include special comment in Podfile. It's a marker after which pods are inserted. That comment of course can be placed multiple times so that they link each target. 

Usecase proposed here is a bit too complex for us, but I guess checking for `pod React` and inserting after it is a good starting point. Just like we insert projects to gradle files.

CC: @Kureev

Reply:
I was also thinking about rewriting the way we inject dependencies to gradle files (it is quite corky now). Although, this implies writing a parser for gradle files which is a bit too much for me. If I'd work on this full time, I'd say it'll take me a couple of weeks to make it properly, so I doubt if it will be worth it. At the same moment of time I completely agree about multiple targets. 

Are you aware of any projects that are capable of constructing / traversing an AST for a Podfile?

Reply:
@Kureev  I did some research when I started work on this. But I couldn't find any library that could parse Podfile (or any Ruby file) in JS.
For me, the whole idea of  `react-native link` is to update project configuration, if it is quite simple. If user starts modifying configuration files by hand, there will be always a case when auto-updating will fail. Or code that update configuration would be quite complex.

@grabbou I might add search for these special comment(s) in file, then adding new pod under it. And if there is no such comment, I might use current approach (as the original idea, is to make it work with ejected expo projects).